hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcCall.java

```diff
@@ -132,4 +132,24 @@ public interface RpcCall extends RpcCallContext {
 
   /** Returns A short string format of this call without possibly lengthy params */
   String toShortString();
+
+  /**
+   * The queue type where the call is located. If {@link RWQueueRpcExecutor} is used, the queue type
+   * can be divided into read, write and scan.
+   */
+  static enum CallQueueType {
```

Comment:
This is an interface for general rpc call, so I'm not sure if it is a good idea to expose the CallQueueType here, as it is related to RpcScheduler, and not all RpcSchedulers have different call queue type, and may be in the future, there could be RpcSchedulers even without a queue?

Reply:
Regarding this issue, I have considered this before. There is no relevant context in the the method `org.apache.hadoop.hbase.ipc.RpcServer#call` to indicate which executor is currently used, so this information can only be obtained based on the incoming parameter `RpcCall call`.  I haven't found the suitable place for CallQueueType yet, maybe I should add a subclass? Do you have any suggestions?